src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,98 @@
-import React from 'react';
-import { NewCommentForm } from '../NewCommentForm';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+import { getPostDetails } from '../../api/posts';
+import { getPostComments, deletePostComment } from '../../api/comments';
+
+import { NewCommentForm } from '../NewCommentForm';
+import { Loader } from '../Loader/Loader';
+
+export const PostDetails = ({ selectedPostId }) => {
+  const [selectedPost, setSelectedPost] = useState(null);
+  const [comments, setComments] = useState(null);
+  const [showComments, setShowComments] = useState(true);
+  
+  useEffect(() => {
+    getPostDetails(selectedPostId)
+      .then(result => setSelectedPost(result))
+
+    updateComents(selectedPostId)
+  }, [selectedPostId]);
+
+  const updateComents = (selectedPostId) => {
+    getPostComments()
+      .then(result => setComments(
+        result.filter(({ postId }) => postId === selectedPostId)
```

Comment:
You can use the endpoint `/comments?postId=${selectedPostId}`
and load comments only for the current post instead of loading all comments and filter them